user-guide.md

```diff
@@ -86,6 +106,45 @@ Make sure you installed JupyterLab Desktop following the steps outlined above in
+- If you install a new JupyterLab extension with UI components, then you will need to create a new session for changes to take effect
+
+# Configuration and data files
+
+JupyterLab Desktop stores user settings, project settings and application data in different locations as JSON files. Below are the storage locations and type of data they contain.
+
+- User settings: `{jlab-desktop-user-data-dir}/settings.json`
+
+  This file contains application settings such as default Python path and theme. These settings can be configured from Settings dialog in the application UI.
+
+- Project settings: `{working-directory}/.jupyter/desktop-settings.json`
+
+  This file contains project (working directory) specific overrides of user settings. Currently only `pythonPath` (which is the Python executable path for the Python environment to use for the working directory) setting can be overridden.
+
+- Application data: `{jlab-desktop-user-data-dir}/app-data.json`
+
+  This file contains data used by the application, e.g. recent sessions list, news feed cache, Python environment list cache.
+
+`{jlab-desktop-user-data-dir}` is OS specific and as below
+
+- `%APPDATA%\jupyterlab-desktop` on Windows
+- `$XDG_CONFIG_HOME/jupyterlab-desktop` or `~/.config/jupyterlab-desktop` on Linux
+- `~/Library/Application Support/jupyterlab-desktop` on macOS
+
+JupyterLab Desktop also uses custom configuration paths for better user experience and to prevent clashes with existing Jupyter installations.
+
+Jupyter config files directory (`JUPYTER_CONFIG_DIR`) is set to `{jlab-desktop-user-data-dir}`. jupyter-server settings, jupyterlab settings, and any other Jupyter configuration are saved to and loaded from this directory. You can change the configuration path by specifying `JLAB_DESKTOP_CONFIG_DIR` environment variable.
+
+JupyterLab workspace data is stored into the working directory, for each folder a new session is started in. This allows restoring open files and UI layout of sessions for different working directories. `{working-directory}/.jupyter/desktop-workspaces` directory is automatically created and used to save and load workspace data for each working directory. You can change this behavior by specifying `JLAB_DESKTOP_WORKSPACES_DIR` environment variable.
+
+## Copying configuration from previous installation
+
+You can transfer settings from previous JupyterLab installations into JupyterLab Desktop by copying them over to the new configuration path `{jlab-desktop-user-data-dir}` by following these steps:
+
+1. Run `jupyter --paths` to determine where your `config` is stored.
+2. Find directory called `lab` in one of these paths.
+3. If there is a `user-settings` directory in `lab`, you can copy the `lab` directory over to the configuration path `{jlab-desktop-user-data-dir}` of JupyterLab Desktop to keep your old JupyterLab settings.
+
+**Warning**: If you copy over settings from an older major version of JupyterLab (e.g. 2.x) those might cause an error on startup.
```

Comment:
```suggestion
**Warning**: If you copy over settings from an older major version of JupyterLab (e.g. 2.x) those might cause an error on startup. Similarly, if you copy settings from a newer (major or minor) version you may see errors.
```

In point releases we often add new settings so this also applies (e.g. 3.6 → 3.5 might error out if someone changed `displayMode`)